pkg/test/framework.go

```diff
@@ -136,7 +135,7 @@ func AddToFrameworkScheme(addToScheme addToSchemeFunc, obj runtime.Object) error
 	Global.RestMapper.Reset()
 	Global.DynamicClient, err = dynclient.New(Global.KubeConfig, dynclient.Options{Scheme: Global.Scheme, Mapper: Global.RestMapper})
 	err = wait.PollImmediate(time.Second, time.Second*10, func() (done bool, err error) {
-		if Global.InCluster {
+		if *Global.SingleNamespace {
 			err = Global.DynamicClient.List(goctx.TODO(), &dynclient.ListOptions{Namespace: os.Getenv(TestNamespaceEnv)}, obj)
```

Comment:
Instead of doing `os.Getenv(TestNamespaceEnv)` wherever it's needed, let's initialize a Namespace field to the framework.
During the framework `setup()` we check this once (if singleNamespace=true) and set Global.Namespace=`<namespace>`.